[++][++][++] lab3 - Karabanov

Roma37 opened the pull request.

Commit:
Karabanovlab3

makometr commented on Karabanov/lab3/Source/ADS3.cpp:
```diff
@@ -0,0 +1,181 @@
+    inline void expError();
+    inline void resize();
+    inline void push(char value);
+    inline char pop();
```
Какой цель служит ключевое слово inline?

makometr commented on Karabanov/lab3/Source/ADS3.cpp:
```diff
@@ -0,0 +1,181 @@
+private:
+    char *ptrStack; 
+    int top;        
+    int maxSize;    
```
поля класса непроинциализированы прилично

makometr commented on Karabanov/lab3/Source/ADS3.cpp:
```diff
@@ -0,0 +1,181 @@
+    void init(){
+        top = 0;
+    }
+};
```
На данные момент класс как захватывающий ресурсы  реализован некорректно. Подумайте, что произойдёт, если попытаться скопировать объекты класса.

makometr reviewed (changes requested):
Тестов, конечно, маловато, ну да ладно

Roma37 committed:
Update ADS3.cpp

Убрал inline,а мне они так понравились...
Конструктор копирования и оператор присваивания теперь есть
